File: sbe/binance/spot_3_2/generated-python/spot_sbe/OptionalVarString.py
"""Generated SBE (Simple Binary Encoding) message codec."""

import struct
from typing import List, Optional, Tuple
from io import BytesIO

class OptionalVarString:
    """Composite type."""

    def __init__(self):
        self.length = 0
        self.var_data = 0

    def encode(self, buffer: BytesIO) -> None:
        """Encode the composite to the buffer."""
        buffer.write(struct.pack('<H', self.length))
        buffer.write(struct.pack('<B', self.var_data))

    def decode(self, buffer: BytesIO) -> None:
        """Decode the composite from the buffer."""
        self.length = struct.unpack('<H', buffer.read(2))[0]
        self.var_data = struct.unpack('<B', buffer.read(1))[0]

    @staticmethod
    def encoded_length() -> int:
        return -1

File: sbe/binance/spot_3_1/generated-php/spot_sbe/OptionalVarString.php
<?php
/**
 * Generated SBE (Simple Binary Encoding) message codec.
 */

class OptionalVarString
{
    public int|float $length = 0;
    public int|float $varData = 0;

    public function encode(): string
    {
        $buffer = '';

        $buffer .= pack('v', $this->length);
        $buffer .= pack('C', $this->varData);

        return $buffer;
    }

    public function decode(string $data, int &$offset = 0): void
    {
        $this->length = unpack('v', substr($data, $offset, 2))[1];
        $offset += 2;
        $this->varData = unpack('C', substr($data, $offset, 1))[1];
        $offset += 1;
    }

    public static function encodedLength(): int
    {
        return -1;
    }
}
